includes/Abstracts/Abstract_Feature.php

```diff
@@ -132,12 +200,19 @@ public function get_description(): string {
 	/**
 	 * Checks if feature is enabled.
 	 *
+	 * Uses injected Feature_Toggles service to check persisted toggle state.
+	 * Falls back to default enabled state if service not available.
+	 *
 	 * @since 0.1.0
 	 *
 	 * @return bool True if enabled, false otherwise.
 	 */
 	final public function is_enabled(): bool {
-		$enabled = $this->enabled;
+		if ( null !== $this->feature_toggles ) {
+			$enabled = $this->feature_toggles->is_feature_enabled( $this->id );
+		} else {
+			$enabled = $this->enabled;
+		}
```

Comment:
I'm a bit confused on the enablement checks. We have:

- `$this->feature_toggles->is_feature_enabled( $this->id );`
- `$this->enabled`
- `apply_filters( "ai_feature_{$this->id}_enabled", $enabled )`

It's hard to know what the source of truth is. Do have needs for this complexity already?